src/lib/flexbox/api/img-srcset.ts

```diff
@@ -0,0 +1,132 @@
+  Renderer,
+  SimpleChanges
+} from '@angular/core';
+import {BaseFxDirective} from './base';
+import {MediaMonitor} from '../../media-query/media-monitor';
+import {ɵgetDOM as getDom} from '@angular/platform-browser';
+
+/**
+ * Directive that injects -in a container <picture> element- <source> elements with media and
+ * srcset attributes.
+ * <source> elemets are sorted according to the related media query : from largest to smallest
+ *
+ * For browsers not supporting the <picture> element, the Picturefill polyfill is still needed.
+ *
+ * @see https://html.spec.whatwg.org/multipage/embedded-content.html#the-picture-element
+ * @see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/picture
+ * @see https://www.html5rocks.com/en/tutorials/responsive/picture-element/
+ * @see https://caniuse.com/#search=picture
+ * @see http://scottjehl.github.io/picturefill/
+ */
+@Directive({
+  selector: `
+  [srcset],
+  [srcset.xs], [srcset.sm], [srcset.md], [srcset.lg], [srcset.xl],
+  [srcset.lt-sm], [srcset.lt-md], [srcset.lt-lg], [srcset.lt-xl],
+  [srcset.gt-xs], [srcset.gt-sm], [srcset.gt-md], [srcset.gt-lg]
+`
+})
+export class ImgSrcsetDirective extends BaseFxDirective implements OnInit, OnChanges {
+
+  /** Reference to injected source elements to be used when there is a need to update their
+   * attributes. */
+  private _registrySourceElements: {[input: string]: any} = {};
+
+  /**
+   * Intercept srcset assignment so we cache the default static value.
+   * When the responsive breakpoint deactivates,it is possible that fallback static
+   * value (which is used to clear the deactivated value) will be used
+   * (if no other breakpoints activate)
+   */
+  @Input('srcset')
+  set srcsetBase(val) {
+    this._cacheInput('srcset', val);
+  }
+
+  /* tslint:disable */
+  @Input('srcset.xs') set srcsetXs(val) {this._cacheInput('srcsetXs', val);}
+  @Input('srcset.sm') set srcsetSm(val) {this._cacheInput('srcsetSm', val);};
+  @Input('srcset.md') set srcsetMd(val) {this._cacheInput('srcsetMd', val);};
+  @Input('srcset.lg') set srcsetLg(val) {this._cacheInput('srcsetLg', val);};
+  @Input('srcset.xl') set srcsetXl(val) {this._cacheInput('srcsetXl', val);};
+
+  @Input('srcset.lt-sm') set srcsetLtSm(val) {this._cacheInput('srcsetLtSm', val);};
+  @Input('srcset.lt-md') set srcsetLtMd(val) {this._cacheInput('srcsetLtMd', val);};
+  @Input('srcset.lt-lg') set srcsetLtLg(val) {this._cacheInput('srcsetLtLg', val);};
+  @Input('srcset.lt-xl') set srcsetLtXl(val) {this._cacheInput('srcsetLtXl', val);};
+
+  @Input('srcset.gt-xs') set srcsetGtXs(val) {this._cacheInput('srcsetGtXs', val);};
+  @Input('srcset.gt-sm') set srcsetGtSm(val) {this._cacheInput('srcsetGtSm', val);};
+  @Input('srcset.gt-md') set srcsetGtMd(val) {this._cacheInput('srcsetGtMd', val);};
+  @Input('srcset.gt-lg') set srcsetGtLg(val) {this._cacheInput('srcsetGtLg', val);};
+
+  /* tslint:enable */
+  constructor(elRef: ElementRef, renderer: Renderer, monitor: MediaMonitor) {
```

Comment:
@benbraou - to account for pending changes in PR #372, please use the `Renderer2` instead of `Renderer`. This should be the last change required before merging. Thx.